test/Dialect/Arc/arc-canonicalizer.mlir

```diff
@@ -491,6 +491,48 @@ hw.module @Repeated_input(in %b: i8, in %e: i8, in %h: i8, in %k: i8, in %en: i1
 // CHECK-NEXT:    hw.output
 // CHECK-NEXT:  }
 
+arc.define @Queue_24_arc(%arg0: i1, %arg1: i1, %arg2: i1, %arg3: i1) -> i124 {
+  %false = hw.constant false
+  %c0_i73 = hw.constant 0 : i73
+  %c0_i46 = hw.constant 0 : i46
+  %0 = comb.concat %c0_i73, %arg0, %arg1, %false, %arg2, %arg3, %c0_i46 : i73, i1, i1, i1, i1, i1, i46
+  arc.output %0 : i124
+}
+arc.define @TLXbar_8_arc() -> i1 {
+  %false = hw.constant false
+  arc.output %false : i1
+}
+arc.define @dummy(%arg0: i124, %arg1: i124) -> i124 {
+  %0 = comb.or %arg0, %arg1 : i124
+  arc.output %0 : i124
+}
+
+hw.module @Repeated_again(in %clock : !seq.clock, in %0 "" : i1) {
+  %1:2 = arc.vectorize (%clock, %clock) : (!seq.clock, !seq.clock) -> (i1, i1) {
+  ^bb0(%arg0: !seq.clock):
+    %4 = arc.state @TLXbar_8_arc() clock %arg0 latency 1 : () -> i1
+    arc.vectorize.return %4 : i1
+  }
+  %2:2 = arc.vectorize (%1#0, %1#1), (%1#0, %1#1), (%0, %0), (%0, %0) : (i1, i1, i1, i1, i1, i1, i1, i1) -> (i124, i124) {
+  ^bb0(%arg0: i1, %arg1: i1, %arg2: i1, %arg3: i1):
+    %4 = arc.call @Queue_24_arc(%arg1, %arg1, %arg3, %arg3) : (i1, i1, i1, i1) -> i124
+    arc.vectorize.return %4 : i124
+  }
+  %3 = arc.state @dummy(%2#0, %3) clock %clock latency 1 : (i124, i124) -> i124
```

Comment:
Nit: any particular reason this `state` (and the arc it's referring to) has to be here or could we just add `%2` as module output?

Reply:
No particular reason for this. I just wanted to use `%2`, I don't know how to use it as a module output, if we don't use it, it will be eliminated by the optimizer.